classes/security/authorization/internal/SubmissionCompletePolicy.inc.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * @file classes/security/authorization/internal/SubmissionCompletePolicy.inc.php
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionCompletePolicy
+ * @ingroup security_authorization_internal
+ *
+ * @brief Class to control access to workflow only for complete submissions
+ *
+ */
+
+ import('lib.pkp.classes.security.authorization.DataObjectRequiredPolicy');
+
+class SubmissionCompletePolicy extends DataObjectRequiredPolicy {
+
+	/**
+	 * Constructor
+	 * 
+	 * @param PKPRequest 	$request 					The PKP core request object
+	 * @param array 		$args 						Request parameters
+	 * @param string 		$submissionParameterName 	The request parameter we expect the submission id in.
+	 * @param string 		$operation 					Optional list of operations for which this check takes effect. If specified, 
+	 * 													operations outside this set will not be checked against this policy
+	 */
+	function __construct($request, &$args, $submissionParameterName = 'submissionId', $operations = null) {
+		parent::__construct(
+			$request,
+			$args,
+			$submissionParameterName,
+			'user.authorization.submission.incomplete.workflowAccessRestrict',
+			$operations
+		);
+
+		AppLocale::requireComponents(LOCALE_COMPONENT_PKP_USER);
+	}
+
+
+	//
+	// Implement template methods from AuthorizationPolicy
+	//
+	/**
+	 * @see DataObjectRequiredPolicy::dataObjectEffect()
+	 */
+	public function dataObjectEffect() {
+		$submissionId = $this->getDataObjectId();
+
+		if ($submissionId === false) {
+			return AUTHORIZATION_DENY;
+		}
+
+		$submissionDao = DAORegistry::getDAO("SubmissionDAO"); /** @var SubmissionDAO $submissionDao */
+		$submission = $submissionDao->getById($submissionId);
+
+		if ($submission->getData('submissionProgress') > 0) {
```

Comment:
We do need to allow access to authors, admins and journal managers to the incomplete submissions. Does it take that into account?

Reply:
Yes , authors or Journal Managers/Editors can access the incomplete submission to complete it . but say an incomplete submission has id `10`. usually the workflow stages should not be accessible for that submission id . but if a user (who has access to workflow stages e.g. with JM/Editor/Admin role) put that submission id in workflow url like `http://ojs-stable-3_3_0.test/index.php/test-01/workflow/index/INCOMPLETE_SUBMISSION_ID/1` , they can initiate the workflow functionality for that incomplete submission .

Reply:
Makes sense!

Reply:
I meant to prevent access to submission wizard for incomplete submission for specific roles but it looks that this isn't needed